lib/application.js

```diff
@@ -71,10 +71,12 @@ app.defaultConfiguration = function defaultConfiguration() {
   var env = process.env.NODE_ENV || 'development';
 
   // default settings
+  this.defaultExtendedQueryParser = false;
   this.enable('x-powered-by');
   this.set('etag', 'weak');
   this.set('env', env);
   this.set('query parser', 'extended');
+  this.defaultExtendedQueryParser = true;
```

Comment:
This is just a question: I see you're setting to `false` a few lines up and then setting `true` here. Is there a particular reason not to just set it one time to `true`? Or, should the first set also be setting to `true` since the default state is that the default query parser is being used?

Side note as well: we may want to either make this property start with an underscore to use the denotation of "private", or perhaps use the same "symbol" tracking used with the trust proxy setting tracking (which was done in order to keep it from displaying during an iteration -- due to lack of symbol support in Node.js 0.10 that 4.x supports).

Reply:
@dougwilson - the idea is to use the tracker such that if the parser is set from the constructor, set it to true, else to false; so that when the middleware is created, the tracker boolean in the tracker will tell us whether an explicit call was present or not.

whether to start with a true or false, looks immaterial to me, I just set it to true, so that the value matches its name.

Also prefixed the name with an underscore to designate its private status.